src/stream/src/executor/top_n/group_top_n.rs

```diff
@@ -387,7 +403,7 @@ mod tests {
                 ActorContext::create(0),
                 order_types,
                 (0, 2),
-                3,
+                1,
```

Comment:
Why to change this? Seems incorrect.

Reply:
We use `order_by_len` to split key into (`order_key|input_pk`), while `order_by_len`  is no used in `WITH_TIES==false`.

Reply:
Ummm I feel confused now. Previously for with_out_ties, `order_by_len` is not needed and the `OrderedRow` is not splitted. Now it's splitted in both cases?

Reply:
Yes, with_out_ties we also need to split `OrderedRow`, and maybe `OrderedRow` will be removed in the future.

Reply:
So we need to update the outdated comment saying `order_by_len` is only used by `WITH_TIES`? BTW, does this change affect the performance of without_ties? 🤔

Reply:
Yes, let's change the comment about `order_by_len `. 
The purpose of this PR is to save cache memory, and there are some costs when serializing, this is by design (save memory with computation, and already done in extreme).  For `without_ties `, no more performance issue.